server-spi-private/src/main/java/org/keycloak/models/utils/RepresentationToModel.java

```diff
@@ -1460,12 +1460,7 @@ private static void updateResources(Set<String> resourceIds, Policy policy, Auth
                 }
 
                 if (!hasResource) {
-                    // for admin permissions remove resource in the FGAP context (if resource is becoming on orphan, we remove the resource from DB)
-                    if (AdminPermissionsSchema.SCHEMA.isAdminPermissionClient(authorization.getRealm(), resourceServer.getId())) {
-                        AdminPermissionsSchema.SCHEMA.removeResource(resourceModel, policy, authorization);
-                    } else {
-                        policy.removeResource(resourceModel);
-                    }
+                    AdminPermissionsSchema.SCHEMA.removeResource(resourceModel, policy, authorization);
```

Comment:
Keeping the logic to remove resources within the schema object.

Reply:
I see, I have no problem to put everything into schema object. Just please note that we are hiding the removal for non-fgap use-cases there.

If you take a look at the code here https://github.com/keycloak/keycloak/blob/ce09061264fed1109a97f8c9a362029fc9e2c050/server-spi-private/src/main/java/org/keycloak/models/utils/RepresentationToModel.java#L1465-L1467

it is now "hidden" inside the schema.

Reply:
let's keep it as is